vueui/src/collections.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * @file
+ * @copyright 2020 Aleksej Komarov
+ * @license MIT
+ */
+
```

Comment:
If I understand it correctly most of the functions here could probably be simplified quite a lot with some modern Object.* calls.

Reply:
Probably. I'm not comfortable enough in JS (especially modern JS) to say, though.